lib/tasks/migrate_trn_to_profiles.rake

```diff
@@ -1,18 +1,26 @@
 namespace :jobseeker_profile do
-  desc "Migrate Teacher Reference Number (TRN) from job_applications to jobseeker_profiles"
+  desc "Migrate Teacher Reference Number (TRN) from job_applications to jobseeker_profiles, creating profiles if necessary"
   task migrate_teacher_reference_number_to_profiles: :environment do
     batch_size = 500
 
-    JobseekerProfile.find_in_batches(batch_size: batch_size) do |profiles|
-      profiles.each do |profile|
-        latest_job_application = JobApplication
-                                   .where(jobseeker_id: profile.jobseeker_id)
-                                   .order(created_at: :desc)
-                                   .first
+    JobApplication
+      .select("DISTINCT job_applications.jobseeker_id, job_applications.teacher_reference_number_ciphertext")
+      .where.not(teacher_reference_number_ciphertext: [nil, ""])
+      .joins("LEFT JOIN jobseeker_profiles ON jobseeker_profiles.jobseeker_id = job_applications.jobseeker_id")
+      .where("jobseeker_profiles.teacher_reference_number_ciphertext IS NULL OR jobseeker_profiles.teacher_reference_number_ciphertext = ''")
+      .find_in_batches(batch_size: batch_size) do |applications|
```

Comment:
would it not be safer to write this in Active Record rather than SQL? Do we not want the *last* job application (by date) rather than an arbitrary one?

small point: you can use find_each rather than find_in_batches - it takes the same parameters and saves you an extra loop.